### PR TITLE
Do not ignore EVP_PKEY_print_public/EVP_PKEY_print_private return values

### PR DESCRIPTION
**Pull request by beldmit**
openssl pkey command ignored the return codes of EVP_PKEY_print_public/EVP_PKEY_print_private

This commit provides the missing checks.

### CODE REVIEW

**Comment by beldmit** [comment]
#6999

**Comment by romen** [comment]
You might want to apply the same treatment also to:
-  [`PEM_write_bio_PUBKEY ` (L189)](https://github.com/openssl/openssl/blob/d6923e12678601191c899789d111aaa36241d2b3/apps/pkey.c#L189)
-  [`PEM_write_bio_PrivateKey_traditional ` (L193)](https://github.com/openssl/openssl/blob/d6923e12678601191c899789d111aaa36241d2b3/apps/pkey.c#L193)
-  [`PEM_write_bio_PrivateKey ` (L197)](https://github.com/openssl/openssl/blob/d6923e12678601191c899789d111aaa36241d2b3/apps/pkey.c#L197)
-  [`i2d_PUBKEY_bio` (L202)](https://github.com/openssl/openssl/blob/d6923e12678601191c899789d111aaa36241d2b3/apps/pkey.c#L202)
-  [`i2d_PrivateKey_bio ` (L205)](https://github.com/openssl/openssl/blob/d6923e12678601191c899789d111aaa36241d2b3/apps/pkey.c#L205)

The only difference would be that these functions all return either 0 (failure) or 1 (success), so it is preferred to style the `if` clause like this:

```c
    if(!foo(args))
        goto err;
```

The functions you already wrapped in `if` clauses instead can return special negative values, so the style you used is fitting.

**Comment by beldmit** [comment]
Thank you. Done.

File: apps/pkey.c [hunk 1]
```diff
@@ -186,23 +186,29 @@ int pkey_main(int argc, char **argv)
...
-                PEM_write_bio_PUBKEY(out, pkey);
+                if (!PEM_write_bio_PUBKEY(out, pkey))
+                    goto end;
             } else {
                 assert(private);
-                if (traditional)
-                    PEM_write_bio_PrivateKey_traditional(out, pkey, cipher,
+                if (traditional) {
+                    if (!PEM_write_bio_PrivateKey_traditional(out, pkey, cipher,
                                                          NULL, 0, NULL,
```
Comment (romen) [inline apps/pkey.c]: Arguments should be aligned

File: apps/pkey.c [hunk 2]
```diff
@@ -186,23 +186,29 @@ int pkey_main(int argc, char **argv)
...
-                                                         passout);
-                else
-                    PEM_write_bio_PrivateKey(out, pkey, cipher,
-                                             NULL, 0, NULL, passout);
+                                                         passout))
+                        goto end;
+                } else {
+                    if (!PEM_write_bio_PrivateKey(out, pkey, cipher,
+                                             NULL, 0, NULL, passout))
```
Comment (romen) [inline apps/pkey.c]: here as well

File: apps/pkey.c (hunk 1 above)
Comment (romen) [reply apps/pkey.c]: I did not check, but if the lines exceed 80 characters, you might need also to truncate the line earlier by moving the arguments on the following lines.

Comment (beldmit) [reply apps/pkey.c]: util/openssl-format-source does not complain here.

Comment (romen) [reply apps/pkey.c]: Perfect, I was just being cautious!

**Comment by beldmit** [comment]
I think it is cherry-pickable to 1.1.0 either.

**Comment by romen** [comment]
> I think it is cherry-pickable to 1.1.0 either.

@beldmit It does not cherry pick cleanly to `OpenSSL_1_1_0-stable`, it would be easier to open a separate PR for backporting to 1.1.0 and obtain the required approval.

I'll merge this shortly in `master`.

**Comment by romen** [comment]
Merged, closing.

Thanks @beldmit for your contribution!

**Pull request closed without merging**